Description: Defines __GNU__ to use the FHS
Author: Scott Howard <showard@debian.org>

Index: rxtx/src/SerialImp.h
===================================================================
--- rxtx.orig/src/SerialImp.h	2011-02-23 23:26:14.955993397 -0500
+++ rxtx/src/SerialImp.h	2011-02-23 23:26:52.684023837 -0500
@@ -147,7 +147,7 @@
 };
 
 /*  Ports known on the OS */
-#if defined(__linux__) || defined(__FreeBSD_kernel__)
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
 /*
 	This is a small hack to get mark and space parity working on older systems
 	https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=147533
