Description: print exception if initialization fails
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=472053
Author: Jan Niehusmann <jan@gondor.com>

Index: rxtx/src/gnu/io/RXTXPort.java
===================================================================
--- rxtx.orig/src/gnu/io/RXTXPort.java	2011-02-03 20:48:21.824924891 -0500
+++ rxtx/src/gnu/io/RXTXPort.java	2011-02-03 20:50:04.058464228 -0500
@@ -86,7 +86,9 @@
 	{
 		try {
 			z = new Zystem();
-		} catch ( Exception e ) {}
+		} catch ( Exception e ) {
+			throw new Error(e.toString());
+		}
 
 		if(debug ) 
 			z.reportln( "RXTXPort {}");
